### ISSUE TITLE
Order of Dataset?

### ISSUE THREAD

**Issue by J1nqianChen** [SuihongSong/GeoModeling_GANSim-2D_Condition_to_Probability_Maps_and_Others#1]
Dear Dr.Song:
Hi! Thanks for your contribution. I'm trying to load your dataset and make (prob.map, facies models) as pairs to serve as training data . But after visulization, I found that the data in 'TrainingData-1r06.tfrecords' and 'TrainingData-2probimages.tfrecords' are not corresponded. Below is the first ten image of these two array. How can I find the correspondence? Thanks!

![image](https://github.com/user-attachments/assets/53e37a86-dfb6-4c15-ad84-b3596f3314b3)

Here is my loading code.
```
def load_geo_data():
    # Facies
    train_path = './GeoData/train/TrainingData-1r06.tfrecords'
    # TrainingData-1r02.tfrecords
    description = {
        'data': 'byte',
        'shape': 'int'
    }

    list_facies = []
    for index, record in enumerate(tfrecord_loader(train_path, index_path=None, description=description)):
        image = decode_record(record)
        image = torch.tensor(image)
        list_facies.append((index, image))


    # Sort by index to maintain original order
    # list_facies.sort(key=lambda x: x[0])
    facies_tensor = torch.stack([x[1] for x in list_facies], dim=0)


    prob_path = './GeoData/train/TrainingData-2probimages.tfrecords'
    list_prob = []
    for index, record in enumerate(tfrecord_loader(prob_path, index_path=None, description=description)):
        image = decode_record(record, data_type='float')
        image = torch.tensor(image)
        list_prob.append((index, image))

    # list_prob.sort(key=lambda x: x[0])
    prob_tensor = torch.stack([x[1] for x in list_prob], dim=0)
    # print(prob_tensor.shape)
    return facies_tensor, prob_tensor
```

**Comment by SuihongSong** [comment]
Hi, 
Thanks for reaching out. 
The dataset (facies models, probability maps, well facies, and global features) can be made either consistent or deliberately inconsistent. Please check the training data preparing file: https://github.com/SuihongSong/GeoModeling_GANSim-2D_Condition_to_Probability_Maps_and_Others/blob/master/Code/Preparing_training_and_test_datasets.ipynb, in section "6 Generate training and test datasets", there are two options for consistent and inconsistent ways. 
Thanks
Suihong

**Comment by J1nqianChen** [comment]
Dear Dr.Song:
Hi！Thanks for your kind reply and sorry to bother again. May I know current training datasets you kindly provided in Zenodo with tfrecords are consistent or not? It puzzles me for a long time..
I just want to figure out whether the tfrecord_loader package is problematic...
As you seen, I want to transfer this project to pytorch enviroment, so I didn't directly utilize tensorflow to load data. 
![image](https://github.com/user-attachments/assets/b495605d-af10-42f9-a8cd-efbe99a50506)

**Comment by SuihongSong** [comment]
Hi Jinqian, 
No. I checked the data this morning. The training facies models and probability maps are not consistent. The facies images you visuallized previously are correct based on my test, but the visuallized probability maps are not. The first four probability maps are as follows. Let me know if you have other questions.
Thanks
Suihong 
![image](https://github.com/user-attachments/assets/5c74d3b5-4257-46c7-83fc-b56d78109e45)

**Comment by J1nqianChen** [comment]
Hi Dr.Song: 
I sincerely appreciate your reply. I have reproduced the probability maps via Gaussian Kernels as you applied to the original facies to ensure the correspondence. But regretfully, I have another question: Are the global features, i.e., the 4rxx.labels file, consistent to the facies? Since they can not be manually generated via facies, it is hard to judge the consistency. Thanks.

Yours best,
Jinqian

**Comment by J1nqianChen** [comment]
I'm sorry. I have checked the preparation code of the datasets. In the inconsistent scenarios, the global features are still consistent with facies. Sorry for the inconvenience.
